pkg/accountmanager/manager.go

```diff
@@ -0,0 +1,74 @@
+package accountmanager
+
+import (
+	"fmt"
+
+	"github.com/Optum/dce/pkg/errors"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/aws/aws-sdk-go/service/sts/stsiface"
+)
+
+// AccountManager manages account resources
+type AccountManager struct {
+	awsConfig           *aws.Config
+	sts                 stsiface.STSAPI
+	adminRoleArn        arn.ARN
+	principalRoleName   string `env:"PRINCIPAL_ROLE_NAME"`
+	principalPolicyName string `env:"PRINCIPAL_POLICY_NAME"`
+	principalRoleArn    arn.ARN
+	principalPolicyArn  arn.ARN
+}
+
+// Setup creates a new session manager struct
+func (am *AccountManager) Setup(adminRoleArn string) error {
```

Comment:
The idea here is a single struct that can act against the child accounts.  It uses awsConfig when setting up clients.  This way credentials are obtained once and automatically rotated without additional configuration.  This should simplify account access and credentials, role gathering.

Reply:
> Setup creates a new session manager struct

Is this comment up to date with the code? (what's a "session manager struct", and where is it created?)